docs/static/_redirects

```diff
@@ -33,6 +37,11 @@ https://material-ui.dev/* https://material-ui.com/:splat 301!
 /css-in-js/* /styles/:splat 301
 /premium-themes* https://themes.material-ui.com/ 301
 
+# link shortener
+/r/styles-instance-warning https://material-ui.com/getting-started/faq#i-have-several-instances-of-styles-on-the-page 307
+/r/caveat-with-refs-guide https://material-ui.com/guides/composition/#caveat-with-refs 307
+/r/pseudo-classes-guide https://material-ui.com/guides/composition/#caveat-with-refs 307
```

Comment:
https://en.wikipedia.org/wiki/List_of_HTTP_status_codes#3xx_Redirection

Since it might change 301 seems like the wrong choice. It should be switched to 301 if it actually points to a specific version of the docs i.e. the url is no longer used in new code versions.